src/cfnlint/helpers.py

```diff
@@ -166,10 +167,63 @@
     'AWS::Redshift::Cluster'
 ]
 
+CACHING_DIR = os.path.join(os.path.dirname(__file__), 'data', 'DownloadsMetadata')
 
-def get_url_content(url):
+
+def get_url_content(url, caching=False):
     """Get the contents of a spec file"""
-    res = urlopen(url)
+    etag = None
+    metadata = {}
+    encoded_url = hashlib.sha256(url.encode()).hexdigest()
+    metadata_filename = os.path.join(CACHING_DIR, encoded_url + '.meta.json')
+
+    # To help with Python2's inability to make HEAD requests
+    make_second_request = True
+
+    if caching:
+        # Load in the cache
+        metadata = get_download_metadata(metadata_filename)
```

Comment:
Suggestion, rename function to `load_cache`

Reply:
I prefer the term metadata because what's being stored isn't the cached file itself, but metadata about the file that gets used to check if we have the latest version. There may be future use cases where we want to store additional metadata about the files that isn't related to caching.

Reply:
Yeah, `load_metadata` sounds better